libV2/process.js

```diff
@@ -0,0 +1,31 @@
+const { EventEmitter } = require('events');
+const os = require('os');
+const { logger } = require('./utils');
+
+
+class Process extends EventEmitter {
+    async setup() {
+        const cleanUpFunc = this.join.bind(this);
+        process.on('SIGINT', cleanUpFunc);
+        process.on('SIGHUP', cleanUpFunc);
+        process.on('SIGQUIT', cleanUpFunc);
+        process.on('SIGTERM', cleanUpFunc);
+        process.on('uncaughtException', error => {
+            logger.error('uncaught exception',
+                { error, stack: error.stack.split(os.EOL) });
+            return cleanUpFunc();
+        });
+        await this._setup();
+    }
+
+    async start() {
+        await this._start();
+    }
+
+    async join() {
+        this.emit('exit');
+        await this._join();
```

Comment:
needs clarification: where are these private methods coming from?

Reply:
I'll stub them out for clarity, these would be overridden by subclasses

Reply:
should check for existence first before calling since there is no guarantee they will be overwritten? or create a noop method?

Reply:
I'll do a noop on the base `Proccess` class so they always exist